emcc.py

```diff
@@ -1325,12 +1326,8 @@ def add_link_flag(i, f):
         options.default_object_extension = '.s'
     elif '-M' in newargs or '-MM' in newargs:
       options.default_object_extension = '.mout' # not bitcode, not js; but just dependency rule of the input file
-
-    if options.output_file:
-      if len(input_files) > 1:
-        exit_with_error('cannot specify -o with -c/-S/-E/-M and multiple source files')
-    else:
-      target = unsuffixed_basename(target) + options.default_object_extension
```

Comment:
We don't need this part anymore? What happens `options.output_file` is `None`?

Reply:
Oh, didn't know this was auto-merge...

Reply:
Indeed we don't need this anymore.   This block deals with "compile only" modes which don't rely on target being set (see `get_object_filename` later in this file).     Only when we perform linking to rely on `target` bring set (IIUC).   I attempt to describe this in PR description.